### DISCUSSION TITLE
How can I update a model from its relations?

### DISCUSSION THREAD

**Discussion by nullndr** [prisma/prisma#11815] [Q&A] [answered]
I have the following models:
```prisma
model Customer {
  id        String       @id @default(cuid())
  email     String       @unique
  firstName  String
  lastName  String
  tags      Tag[]
  user      DiscordUser? @relation(fields: [userId], references: [id])
  userId    String?      @unique
}

model Tag {
  id        String            @id @default(cuid())
  text      String            @unique
  customers Customer[]
  roles     DiscordRole[]
}

model DiscordUser {
  id            String            @id
  email         String
  username      String
  discriminator String
  roles         DiscordRole[]
  customer      Customer?
  token         DiscordUserToken?

  @@unique([username, discriminator])
}

model DiscordRole {
  id    String        @id
  name  String        @unique
  users DiscordUser[]
  tags  Tag[]
}
```
When I add an association between a `Tag` record and a `DiscordRole` record, I need to update all `DiscordUser`, with the `DiscordRole` record, associated at `Customer` that have the tag.

For example, if I have the customer `Foo`, with the tag `FooTag`, associated at the discord user `Bar`, when I add an association between `FooTag` and `BarRole`, I need to update `Bar` with the `BarRole`.

Until now, I have successfully connected the `Tag` with the `DiscordRole`:

```typescript
await db.tag.update({
    where: { text: input.tag },
    data: {
      roles: {
        connect: { name: input.role },
      },
    },
  });
```

And I've tried to update the `DiscordUser` model separately, with:

```typescript
await db.discordUser.updateMany({
    where: {
      customer: {
        tags: {
          some: {
            text: {
              equals: input.role
            }
          }
        }
      }
    },
    data: {
      ????
    }
  })
```

But with no success, also I would like to have only one query, not more fragmented.

**Comment by pantharshit00** [comment, marked as answer]
Hello @nullndr , our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

Right now you can't update relations using `updateMany`: https://github.com/prisma/prisma/issues/3143. So you will need to query the record first in order to update it:
```ts
  const dUser = await prisma.discordUser.findFirstOrThrow({
    where:{
      customer:{
        tags:{
          some:{
            text:{
              equals:input.role
            }
          }
        }
      }
    }
  })

  await prisma.discordUser.update({
    where: {
      id: dUser.id
    },
    data: {
      roles:{
        connect:{
          name:"BarRole"
        }
      }
    }
  })
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
